src/packages/admin-ui-components/src/detail-panel/component.tsx

```diff
@@ -253,6 +268,22 @@ export const DetailPanel = () => {
 		try {
 			let result: FetchResult;
 			if (id && !isNew) {
+				// Update an existing entity
+				if (formValues.uploadUrl && formValues.file) {
+					await uploadFileToSignedURL(values.uploadUrl, values.file);
+					// remove the uploadUrl and file from the values. These are set in the <ImageField> on upload file
+					delete values.uploadUrl;
+					delete values.file;
+					// delete the value where the name is the field that has the image or media type
+					delete values[selectedEntity.fields.find((field) => field.type === 'Image')?.name ?? ''];
+					delete values[selectedEntity.fields.find((field) => field.type === 'Media')?.name ?? ''];
+				}
+
+				// if uploadUrl and downloadUrl are there, remove them because theyre not on SubmissionCreateOrUpdateInput
+				if (values.uploadUrl === null) delete values.uploadUrl;
+				if (values.downloadUrl === null || values.downloadUrl === '') delete values.downloadUrl;
```

Comment:
This wont always be `downloadUrl`, we need to make this generic.

Reply:
Will rename the entity downloadUrl to something else to be clearer.